onnxruntime/core/framework/tensorprotoutils.h

```diff
@@ -75,6 +75,11 @@ ONNXTensorElementDataType GetTensorElementType(const ONNX_NAMESPACE::TensorProto
 template <size_t alignment>
 common::Status GetSizeInBytesFromTensorProto(const ONNX_NAMESPACE::TensorProto& tensor_proto, size_t* out);
 
+// Given a tensor proto with external data obtain a pointer to the data and its length.
+// The ext_data_deleter argument is updated with a callback that owns/releases the data.
+Status GetExtDataFromTensorProto(const Env& env, const ORTCHAR_T* model_path, const ONNX_NAMESPACE::TensorProto& tensor_proto,
```

Comment:
FYI, the OrtCallback class was added just for this purpose.

Reply:
Indeed, it handled the problem here neatly. However, although this `OrtValue`'s deleter owns the external data wrapped by its `Tensor` object the latter still requires an `IAllocator` pointer in its constructor to release that same data. To prevent that second release the `GetExtDataFromTensorProto` function uses an instance of a do-nothing allocator (`ExtDataAllocator`). This bit could be removed if the `Tensor` class allowed a nullptr and treated that as a do-nothing special case.

Reply:
We can safely assume the allocator will live longer than SessionState.